src/Files.App/Views/Layouts/BaseLayoutPage.cs

```diff
@@ -903,22 +903,28 @@ private async Task AddShellMenuItemsAsync(List<ContextMenuFlyoutItemViewModel> s
 				}
 			}
 
-			// Add items to main shell submenu
-			mainShellMenuItems.Where(x => x.LoadSubMenuAction is not null).ForEach(async x =>
-			{
-				await x.LoadSubMenuAction();
+			// Filter mainShellMenuItems that have a non-null LoadSubMenuAction
+			var mainItemsWithSubMenu = mainShellMenuItems.Where(x => x.LoadSubMenuAction is not null);
 
-				ShellContextFlyoutFactory.AddItemsToMainMenu(mainItems, x);
-			});
+			// Load main submenus asynchronously
+			await Task.WhenAll(mainItemsWithSubMenu.Select(item => item.LoadSubMenuAction()));
```

Comment:
`ShellContextFlyoutFactory.AddItemsToMainMenu(mainItems, item);` should also be called in `Task.WhenAll` like overflow items.